Vector backend

rebcabin opened the pull request.

certik commented:
Once this is ready for a review, please click on the "Ready for review" button.

Shaikh-Ubaid commented:
I think the `.idea` folder in this PR should not be commited. I think it is a temporary/utility folder created by IntelliJ Idea (similar to the `.vscode` folder that VSCode creates).

certik commented:
I marked it as draft, so that we don't accidentally merge it.

rebcabin commented:
I fixed it in my branch and added .idea to .gitignore

```
(lp) ┌─(~/Documents/GitHub/lpython)─────────────────────────────────────────────────────────(brian@MacBook-Pro:s001)─┐
└─(09:22:54 on vector-backend ✖ ✹)──> git commit -am "remove .idea and add .idea to .gitignore"   ──(Mon,Feb12)─┘
[vector-backend eb2af0103] remove .idea and add .idea to .gitignore
 8 files changed, 1 insertion(+), 71 deletions(-)
 delete mode 100644 .idea/.gitignore
 delete mode 100644 .idea/customTargets.xml
 delete mode 100644 .idea/lpython.iml
 delete mode 100644 .idea/misc.xml
 delete mode 100644 .idea/modules.xml
 delete mode 100644 .idea/tools/External Tools.xml
 delete mode 100644 .idea/vcs.xml
(lp) ┌─(~/Documents/GitHub/lpython)─────────────────────────────────────────────────────────(brian@MacBook-Pro:s001)─┐
└─(09:23:14 on vector-backend)──> git push                                                        ──(Mon,Feb12)─┘
Enumerating objects: 5, done.
Counting objects: 100% (5/5), done.
Delta compression using up to 12 threads
Compressing objects: 100% (3/3), done.
Writing objects: 100% (3/3), 320 bytes | 320.00 KiB/s, done.
Total 3 (delta 2), reused 0 (delta 0), pack-reused 0
remote: Resolving deltas: 100% (2/2), completed with 2 local objects.
To https://github.com/rebcabin/lpython.git
   6fdc4b32c..eb2af0103  vector-backend -> vector-backend

```

CaMo111 commented:
bookmark for writing c compiler
